lib/_stream_readable.js

```diff
@@ -40,20 +40,18 @@ const kProxyEvents = ['error', 'close', 'destroy', 'pause', 'resume'];
 function prependListener(emitter, event, fn) {
   // Sadly this is not cacheable as some libraries bundle their own
   // event emitter implementation with them.
-  if (typeof emitter.prependListener === 'function') {
+  if (typeof emitter.prependListener === 'function')
     return emitter.prependListener(event, fn);
-  } else {
-    // This is a hack to make sure that our error handler is attached before any
-    // userland ones.  NEVER DO THIS. This is here only because this code needs
-    // to continue to work with older versions of Node.js that do not include
-    // the prependListener() method. The goal is to eventually remove this hack.
-    if (!emitter._events || !emitter._events[event])
-      emitter.on(event, fn);
-    else if (Array.isArray(emitter._events[event]))
-      emitter._events[event].unshift(fn);
-    else
-      emitter._events[event] = [fn, emitter._events[event]];
-  }
+  // This is a hack to make sure that our error handler is attached before any
+  // userland ones.  NEVER DO THIS. This is here only because this code needs
+  // to continue to work with older versions of Node.js that do not include
+  // the prependListener() method. The goal is to eventually remove this hack.
+  else if (!emitter._events || !emitter._events[event])
```

Comment:
Nit: this could be an `if` instead of `else if` as there is a `return` statement in the previous `if` body.

Reply:
Nice catch! I'll update this later today.

Funny enough, there's actually a rule I want to introduce that deals with that but it requires changing more files (around 80) so I wanted to keep it for a separate PR in case there was push back.

Reply:
Force pushed to address this.

Reply:
May be a case for Code And Learn (https://github.com/nodejs/code-and-learn/issues/72#issuecomment-337512194)